### PR TITLE
Reactivate coverage tracking for dartdoc via coveralls

### PR DESCRIPTION
**Pull request by jcollins-g**
It's been more than 3 years since coverage worked on dartdoc, but with this PR coverage tracking returns!

https://coveralls.io/github/dart-lang/dartdoc?branch=coveralls for the coverage result for this branch, with this PR.  It compares our coverage to what it was on master, here: https://coveralls.io/builds/11445981

That was the last update from August 10, 2015.

To make this work some hacks were needed.  dart-lang/sdk#31747 meant we had to avoid use of 'exit()' in dart, and dart-lang/coverage#239 meant we had to fork a modified version of the 'format_coverage.dart' binary into dartdoc to get around the type error.    I don't use dart_coveralls here because I'd either have to run the tests twice or lose parallelization in dartdoc's custom test runner.

### CODE REVIEW

**Comment by pq** [comment]
> To make this work some hacks were needed.

Do you feel like there would be any value documenting this somewhere?  Or is `dartdoc` too much of it's own odd thing?

**Comment by jcollins-g** [comment]
I don't think dartdoc is that odd for a complex project with many tests; it's just that 1) we don't have good support for layering things like coverage runs into the default test runner and 2) default test runners don't easily support parallel execution.  So the bits we have only work for trivial cases, or cases where you don't mind spending a lot of cloud resources, or you have to roll your own like what dartdoc does. 

Maybe it would be nice if these issues were addressed in a more comprehensive way, though.

File: pubspec.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ dev_dependencies:
   build: ^1.0.1
   build_runner: ^1.0.0
   build_version: ^1.0.0
+  coverage: any
```
Comment (devoncarew) [inline pubspec.yaml]: Are you intending to let this float?

File: tool/format_coverage.dart [hunk 2]
```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
```
Comment (devoncarew) [inline tool/format_coverage.dart]: This is an alternative front end to package:coverage (to better support merging results from multiple runs)? If so, perhaps add a file or class comment to that effect.

File: pubspec.yaml (hunk 1 above)
Comment (jcollins-g) [reply pubspec.yaml]: Yes.

File: tool/format_coverage.dart (hunk 2 above)
Comment (jcollins-g) [reply tool/format_coverage.dart]: this is just a temporary fork as I mentioned above.  with the next release of coverage, i will constrain version and eliminate the fork.

Comment (devoncarew) [reply tool/format_coverage.dart]: gotcha

**Pull request merged**